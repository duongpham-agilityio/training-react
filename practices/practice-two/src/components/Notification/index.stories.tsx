import { Meta, Story } from '@storybook/react';

import { Notification, NotificationProps } from 'components';

const Template: Story<NotificationProps> = (args) => <Notification {...args} />;
export const Success = Template.bind({});
export const Error = Template.bind({});
export const Warning = Template.bind({});
export const Info = Template.bind({});

Success.args = {
  title: 'Your actions executed successfully!',
  message: 'A book has been removed from the system, this action can not undo.',
};

Error.args = {
  title: 'Your actions executed successfully!',
  message: 'A book has been removed from the system, this action can not undo.',
  type: 'error',
};

Warning.args = {
  title: 'Your actions executed successfully!',
  message: 'A book has been removed from the system, this action can not undo.',
  type: 'warning',
};

Info.args = {
  title: 'Your actions executed successfully!',
  message: 'A book has been removed from the system, this action can not undo.',
  type: 'info',
};

export default {
  title: 'Practice/components/Notification',
  component: Notification,
  decorators: [
    (Story) => (
      <div
        className=""
        style={{
          margin: '20px',
        }}
      >
        <Story />
      </div>
    ),
  ],
} as Meta;
