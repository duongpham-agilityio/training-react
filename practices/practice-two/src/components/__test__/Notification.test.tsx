import { render } from '@testing-library/react';

// Mocks
import './mocks/image.test';

// Components
import Notification, { NotificationProps } from 'components/Notification';

const setup = (props: NotificationProps) => render(<Notification {...props} />);

describe('Notification component', () => {
  const mockProps: NotificationProps = {
    title: 'Success',
    message: 'Send request success',
  };

  it('Match to snapshot', () => {
    const { container } = setup(mockProps);

    expect(container).toMatchSnapshot();
  });
});
