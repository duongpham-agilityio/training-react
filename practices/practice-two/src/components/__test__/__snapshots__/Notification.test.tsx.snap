// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component Match to snapshot 1`] = `
<div>
  <section
    class="undefined undefined undefined"
  >
    <div>
      <h2
        class="undefined undefined "
      >
        Success
      </h2>
      <p
        data-testid="notify-message"
      >
        Send request success
      </p>
    </div>
    <button
      class="undefined undefined
  undefined undefined
  undefined "
      type="button"
    >
      <img
        alt="This is the left icon for the button"
        height="25px"
        src="assets/icons/x-letter.svg"
        width="22px"
      />
    </button>
  </section>
</div>
`;
